internal/castai/castai.go

```diff
@@ -159,10 +166,14 @@ func (c *client) SendDelta(ctx context.Context, clusterID string, delta *Delta)
 		Jitter:   0.2,
 		Steps:    3,
 	}
+	numAttempts := 0
 	err = wait.ExponentialBackoffWithContext(ctx, backoff, func() (done bool, err error) {
+		numAttempts++
+		log = log.WithField("attempts", numAttempts)
```

Comment:
we're overriding the `log` here, not sure if it's okay, shouldn't this be `:=` ?

Reply:
yeah, intentional, updated log will contain the amount of upload attempts there were with downstream logging.